tests/components/fyta/test_config_flow.py

```diff
@@ -190,3 +191,39 @@ async def test_reauth(
     assert entry.data[CONF_PASSWORD] == "other_password"
     assert entry.data[CONF_ACCESS_TOKEN] == ACCESS_TOKEN
     assert entry.data[CONF_EXPIRATION] == EXPIRATION
+
+
+async def test_dhcp_discovery(
+    hass: HomeAssistant, mock_fyta_connector: AsyncMock, mock_setup_entry: AsyncMock
+) -> None:
+    """Test DHCP discovery flow."""
+
+    service_info = DhcpServiceInfo(
+        hostname="FYTA HUB",
+        ip="1.2.3.4",
+        macaddress="aabbccddeeff",
+    )
+
+    result = await hass.config_entries.flow.async_init(
```

Comment:
From here on the code is exactly the same as in the test_user_flow right? Let's put that in a helper function to avoid duplicate code